hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java

```diff
@@ -408,17 +420,26 @@ private long init(List<DatanodeStorageReport> reports) {
     for(DatanodeStorageReport r : reports) {
       final DDatanode dn = dispatcher.newDatanode(r.getDatanodeInfo());
       final boolean isSource = Util.isIncluded(sourceNodes, dn.getDatanodeInfo());
+      final boolean isExcludedSource = Util.isIncluded(sourceNodes, dn.getDatanodeInfo());
+      final boolean isTarget = Util.isIncluded(targetNodes, dn.getDatanodeInfo());
+      final boolean isExcludedTarget = Util.isIncluded(targetNodes, dn.getDatanodeInfo());
```

Comment:
Might be simpler to extend these conditions to check the exclusion rather than make every `if` check below more complex

Reply:
Updated